lib/internal/module.js

```diff
@@ -44,3 +48,27 @@ function stripBOM(content) {
   }
   return content;
 }
+
+exports.builtinLibs = ['assert', 'buffer', 'child_process', 'cluster',
+  'crypto', 'dgram', 'dns', 'domain', 'events', 'fs', 'http', 'https', 'net',
+  'os', 'path', 'punycode', 'querystring', 'readline', 'stream',
+  'string_decoder', 'tls', 'tty', 'url', 'util', 'v8', 'vm', 'zlib'];
+
+function addBuiltinLibsToObject(object) {
+  // Make built-in modules available directly (loaded lazily).
+  exports.builtinLibs.forEach((name) => {
+    Object.defineProperty(object, name, {
+      get: () => {
+        const lib = require(name);
+        object[name] = lib;
```

Comment:
@thefourtheye Nah, it’s a nice but weird trick. The setter below `delete`s the property before re-assigning it, which disables the setter/getter themselves. So only the first assignment or read of this variable uses this mechanism, and nothing gets overridden.
I was a bit confused at first, too, so I guess it’s a good idea to add a comment for explanation. :smile:

(This code has essentially only been moved from `lib/repl.js`, btw)